src/evict/evict_lru.c

```diff
@@ -2638,6 +2638,39 @@ __verbose_dump_cache_single(WT_SESSION_IMPL *session,
 	return (0);
 }
 
+/*
+ * __verbose_dump_cache_apply --
+ *	Apply dumping cache for all the dhandles.
+ */
+static int
+__verbose_dump_cache_apply(WT_SESSION_IMPL *session,
+    uint64_t *total_bytesp, uint64_t *total_dirty_bytesp)
+{
+	WT_CONNECTION_IMPL *conn;
+	WT_DATA_HANDLE *dhandle;
+	WT_DECL_RET;
+
+	conn = S2C(session);
+	for (dhandle = NULL;;) {
+		WT_DHANDLE_NEXT(session, dhandle, &conn->dhqh, q);
+		if (dhandle == NULL)
+			break;
+
+		/* Skip if the tree is marked discarded by another thread. */
+		if (dhandle->type != WT_DHANDLE_TYPE_BTREE ||
+		    !F_ISSET(dhandle, WT_DHANDLE_OPEN) ||
+		    F_ISSET(dhandle, WT_DHANDLE_DISCARD))
+			continue;
+
+		WT_WITH_DHANDLE(session, dhandle,
+		    ret = __verbose_dump_cache_single(
+		    session, total_bytesp, total_dirty_bytesp));
+		if (ret != 0)
+			break;
```

Comment:
As @bvpvamsikrishna this would be better as a WT_RET(ret), then the return at the end of the function can return 0.